lib/Dialect/FIRRTL/Import/FIRParserAsserts.cpp

```diff
@@ -1,622 +0,0 @@
-  ex.emitError() << "must be `formalOnly` or `unrOnly`";
-  return std::nullopt;
-}
-
-/// Parse a string into a `PredicateModifier`.
-static std::optional<PredicateModifier>
-parsePredicateModifier(const ExtractionSummaryCursor<StringRef> &ex) {
-  if (ex.value == "noMod")
-    return PredicateModifier::NoMod;
-  else if (ex.value == "trueOrIsX")
-    return PredicateModifier::TrueOrIsX;
-  ex.emitError() << "must be `noMod` or `trueOrIsX`";
-  return std::nullopt;
-}
-
-/// Check that an assertion "format" is one of the admissible values, or report
-/// an error.
-static std::optional<StringRef>
-parseAssertionFormat(const ExtractionSummaryCursor<StringRef> &ex) {
-  if (ex.value == "sva" || ex.value == "ifElseFatal")
-    return ex.value;
-  ex.emitError() << "must be `sva` or `ifElseFatal`";
-  return std::nullopt;
-}
-
-/// Chisel has a tendency to emit complex assert/assume/cover statements encoded
-/// as print operations with special formatting and metadata embedded in the
-/// message literal. These always reside in a when block of the following form:
-///
-///     when invCond:
-///       printf(clock, UInt<1>(1), "...[verif-library-assert]...")
-///       stop(clock, UInt<1>(1), 1)
-///
-/// Depending on the nature the verification operation, the `stop` may be
-/// optional. The Scala implementation simply removes all `stop`s that have the
-/// same condition as the printf.
-FailureOr<bool> circt::firrtl::foldWhenEncodedVerifOp(PrintFOp printOp) {
-  auto *context = printOp.getContext();
-  auto whenStmt = dyn_cast<WhenOp>(printOp->getParentOp());
-
-  // If the parent of printOp is not when, printOp doesn't encode a
-  // verification.
-  if (!whenStmt)
-    return false;
-
-  // The when blocks we're interested in don't have an else region.
-  if (whenStmt.hasElseRegion())
-    return false;
-
-  // The when blocks we're interested in contain a `PrintFOp` and an optional
-  // `StopOp` with the same clock and condition as the print.
-  Block &thenBlock = whenStmt.getThenBlock();
-  auto opIt = std::next(printOp->getIterator());
-  auto opEnd = thenBlock.end();
-
-  // Detect if we're dealing with a verification statement, and what flavor of
-  // statement it is.
-  auto fmt = printOp.getFormatString();
-  VerifFlavor flavor;
-  if (fmt.contains("[verif-library-assert]"))
-    flavor = VerifFlavor::VerifLibAssert;
-  else if (fmt.contains("[verif-library-assume]"))
-    flavor = VerifFlavor::VerifLibAssume;
-  else if (fmt.contains("[verif-library-cover]"))
-    flavor = VerifFlavor::VerifLibCover;
-  else if (fmt.consume_front("assert:"))
-    flavor = VerifFlavor::Assert;
-  else if (fmt.consume_front("assume:"))
-    flavor = VerifFlavor::Assume;
-  else if (fmt.consume_front("cover:"))
-    flavor = VerifFlavor::Cover;
-  else if (fmt.consume_front("assertNotX:"))
-    flavor = VerifFlavor::AssertNotX;
-  else if (fmt.starts_with("Assertion failed"))
-    flavor = VerifFlavor::ChiselAssert;
-  else
-    return false;
```

Comment:
Not sure it's worth doing but how about extracting this part and emitting warnings for PrintOp with such format string? Though there might be false-alarm but I feel it's better than silently dropping assertions. It would be not that bad to inspect format string at parsing.